### PR TITLE
Skip trivia in parameter names

### LINKED ISSUE #61: Align whitespace and comment handling in collect_parameter_name with parse_type_expr
There is an inconsistency in how whitespace and comments are handled between the `parse_type_expr` and `collect_parameter_name` functions. Currently, `parse_type_expr` filters out whitespace and comments, but `collect_parameter_name` does not. This can result in parameter names containing unwanted whitespace or comments, which may lead to bugs or unexpected behavior.

**Suggested Fix:**
Update `collect_parameter_name` to skip whitespace and comment nodes, aligning its behavior with `parse_type_expr`. For example:

```rust
NodeOrToken::Node(n) => {
    // Skip whitespace and comment nodes for consistency with parse_type_expr
    let text = n.text();
    if n.kind() == SyntaxKind::T_COMMENT || text.chars().all(char::is_whitespace) {
        iter.next();
        continue;
    }
    buf.push_str(text);
    iter.next();
}
```

**Action Items:**
- Update `collect_parameter_name` to filter out whitespace and comments as shown above.
- Add or update tests to ensure parameter names are parsed without unwanted whitespace or comments.
- Review related code for similar inconsistencies.

---

I created this issue for @leynos from https://github.com/leynos/ddlint/pull/57#discussion_r2187855295.

<details>
<summary>Tips and commands</summary>

#### Getting Help

- [Contact our support team](mailto:support@sourcery.ai) for questions or feedback.
- Visit our [documentation](https://docs.sourcery.ai) for detailed guides and information.
- Keep in touch with the Sourcery team by following us on [X/Twitter](https://x.com/SourceryAI), [LinkedIn](https://www.linkedin.com/company/sourcery-ai/) or [GitHub](https://github.com/sourcery-ai).

</details>

### PR DESCRIPTION
**Pull request by leynos**
## Summary
- ignore whitespace and comments while collecting parameter names
- exercise parameter name parsing against comments and excess spacing
- document that helper functions skip trivia

closes #61

## Testing
- `make fmt`
- `make lint`
- `make test`
- `make markdownlint`
- `make nixie` *(fails: Error running command bun x --bun @mermaid-js/mermaid-cli)*

------
https://chatgpt.com/codex/tasks/task_e_689ef898b0348322b82472f9263d2332

## Summary by Sourcery

Skip whitespace and comments when collecting parameter names and update related documentation and tests

Enhancements:
- Make collect_parameter_name and parse_type_expr ignore whitespace and comment nodes during parsing

Documentation:
- Document that helper functions now skip whitespace and comment nodes in function-parsing-design

Tests:
- Add test cases for function parameters with inline comments and excess spacing

Chores:
- Close issue #61

### CODE REVIEW

File: src/parser/ast/parse_utils/type_parsing.rs [hunk 1]
```diff
@@ -349,17 +349,23 @@ where
                     track_span_for_element(e, &mut start_pos, &mut end_pos);
                     break;
                 }
+                SyntaxKind::T_WHITESPACE | SyntaxKind::T_COMMENT => {
```
[the remark this answers is not in the document]
Comment (leynos) [reply src/parser/ast/parse_utils/type_parsing.rs]: Has this now been resolved?

File: src/parser/ast/parse_utils/type_parsing.rs [hunk 2]
```diff
@@ -143,8 +143,15 @@ where
     let mut ctx = TokenParseContext::new(&mut buf, &mut stack);
 
     while let Some(e) = iter.peek() {
+        if is_trivia(e) {
+            iter.next();
+            continue;
+        }
```
[the remark this answers is not in the document]
Comment (leynos) [reply src/parser/ast/parse_utils/type_parsing.rs:149]: @coderabbitai Please suggest a fix for this issue and supply a prompt for an AI coding agent to enable it to apply the fix.

File: src/parser/ast/parse_utils/type_parsing.rs [hunk 3]
```diff
@@ -191,6 +169,63 @@ where
...
+            iter.next();
+            return false;
+        }
+        return true;
+    }
+
+    push(token, ctx);
+    iter.next();
+    false
+}
```
[the remark this answers is not in the document]
Comment (leynos) [reply src/parser/ast/parse_utils/type_parsing.rs:227]: @coderabbitai Please suggest a fix for this issue and supply a prompt for an AI coding agent to enable it to apply the fix.
